src/diskquota.c

```diff
@@ -163,6 +166,40 @@ diskquota_launcher_shmem_size(void)
 	size = add_size(size, mul_size(MAX_NUM_MONITORED_DB, sizeof(struct DiskquotaDBEntry))); // hidden memory for dbArray
 	return size;
 }
+
+static bool
+is_altering_extension(void)
+{
+	if (ActivePortal == NULL) return false;
+	/* QD: When the sourceTag is T_AlterExtensionStmt, then return true */
+	if (ActivePortal->sourceTag == T_AlterExtensionStmt) return true;
+
+	/*
+	 * QE: The sourceTag won't be T_AlterExtensionStmt, we should check the sourceText.
+	 * If the sourceText contains 'alter extension diskquota update', we consider it is
+	 * a alter extension query.
+	 */
+	char *query = asc_tolower(ActivePortal->sourceText, strlen(ActivePortal->sourceText));
+	char *pos;
+	bool  match = true;
+
+	pos = strstr(query, "alter");
+	if (pos)
+		pos = strstr(pos, "extension");
+	else
+		match = false;
+	if (pos)
+		pos = strstr(pos, "diskquota");
+	else
+		match = false;
+	if (pos)
+		pos = strstr(pos, "update");
+	else
+		match = false;
+	pfree(query);
+	return match;
```

Comment:
This function is incorrect if it is necessary that `ActivePortal->sourceText` must contain word `update`. This word is not checked at this function.

Reply:
> This word is not checked at this function.

Why?

Reply:
After save `pos` at line 192 there is check `pos` is not null at line 195. This is check that query contains `diskquota`. If it is true, `match` is not change to `false`. Next saving pointer of `update` to `pos` does not affect to variable `match`. And `true` will be returned even if `strstr(pos, "update");` returns `NULL`.

Reply:
Can you explain step by step?
```
182: query = "alter extension diskquota update"
184: match = true
186: pos = "alter extension diskquota update"
188: pos = "extension diskquota update"
192: pos = "diskquota update"
196: pos = "update"
200: match = true
```

Reply:
If you type another word instead of `update` you will get `match = true` but expect `match = false`.

Reply:
Exactly! If last word is not update then
```
196: pos = NULL
```
They missed another check
```c

	if (pos)
		pos = strstr(pos, "update");
	else
		match = false;
	if (!pos)
		match = false;
```